src/ol/source/TileDebug.js

```diff
@@ -36,32 +43,87 @@ class TileDebug extends ImageTile {
      */
     options = options || {};
     const template = options.template || 'z:{z} x:{x} y:{y}';
+    const source = options.source;
 
     super({
       transition: 0,
-      projection: options.projection,
-      tileGrid: options.tileGrid,
-      wrapX: options.wrapX !== undefined ? options.wrapX : true,
-      zDirection: options.zDirection,
-      loader: (z, x, y, loaderOptions) => {
+      wrapX:
+        options.wrapX !== undefined
+          ? options.wrapX
+          : source !== undefined
+            ? source.getWrapX()
+            : undefined,
+    });
+
+    const setReady = () => {
+      this.projection =
+        options.projection !== undefined
+          ? getProjection(options.projection)
+          : source !== undefined
+            ? source.getProjection()
+            : this.projection;
+      this.tileGrid =
+        options.tileGrid !== undefined
+          ? options.tileGrid
+          : source !== undefined
+            ? source.getTileGrid()
+            : this.tileGrid;
+      this.zDirection =
+        options.zDirection !== undefined
+          ? options.zDirection
+          : source !== undefined
+            ? source.zDirection
+            : this.zDirection;
+
+      if (source instanceof GeoTIFF) {
+        this.transformMatrix = source.getTransformMatrix();
+      }
```

Comment:
The instanceof check should check for `DataTile` instead of `GeoTIFF`, and a copy of the member property instead of the getter should be used when assigning `this.transformMatrix`.